code/datums/elements/_element.dm

```diff
@@ -0,0 +1,28 @@
+/datum/element
+	var/element_flags = NONE
+
+/datum/element/proc/Attach(datum/target)
+	if(type == /datum/element)
+		return ELEMENT_INCOMPATIBLE
+	if(element_flags & ELEMENT_DETACH)
+		RegisterSignal(target, COMSIG_PARENT_QDELETING, .proc/Detach)
+
+/datum/element/proc/Detach(datum/source, force)
+
+/datum/element/Destroy(force)
+	if(!force)
+		return QDEL_HINT_LETMELIVE
+	SSdcs.elements_by_type -= type
+	return ..()
+
+//DATUM PROCS
+
+/datum/proc/AddElement(eletype, ...)
+	var/datum/element/ele = SSdcs.GetElement(eletype)
+	args[1] = src
+	if(ele.Attach(arglist(args)) == ELEMENT_INCOMPATIBLE)
+		CRASH("Incompatible [eletype] assigned to a [type]! args: [json_encode(args)]")
+
+/datum/proc/RemoveElement(eletype)
+	var/datum/element/ele = SSdcs.GetElement(eletype)
+	ele.Detach(src)
```

Comment:
I'm not seeing any evidence that this actually unregisters signals

Reply:
Why would it?

Reply:
You add the cleaning element to something, then remove it, but it keeps cleaning

Reply:
Oh you mean that particular element should, yeah this RemoveElement proc was an after thought as nothing actually removes them in what I was testing

Reply:
It should either be removed or work correctly